broker/proto/broker.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package neon_broker;
+
+service NeonBroker {
+    // Subscribe to safekeeper updates.
+    rpc SubscribeSafekeeperInfo(SubscribeSafekeeperInfoRequest) returns (stream SafekeeperTimelineInfo) {};
```

Comment:
> This doesn't scale well, because it's not convenient to keep N (number of types) active subscription streams for different types,

It seems okay to me given that underlying code is the same (so need to replace SafekeeperTimelineInfo with template argument), not too much boilerplate and have type safety. But what you are describing is also possible.